### PR TITLE
Makes use of lightweight stateless function component syntax for leaf…

### PR DESCRIPTION
**Pull request by gjjones**
Few parts to this:

1: whitespace.
2: updates ```react-redux``` package to clean up a warning that was introduced by...
3: dumb components now use the [stateless function components](http://tylermcginnis.com/functional-components-vs-stateless-functional-components-vs-stateless-components/).

Switching to the stateless function syntax will promote developers to not use state as the hammer(solution) for their nail(problem).

Also, is there a good reason that we're not ignoring ```dist/```? This is something we inherited from the boilerplate, and I'd rather not keep it. I can't think of a good reason to, so please fill me in if I'm being dense.

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
+import React from 'react';
 
 import '!style!css!less!./style.less'
 
 import classNames from 'classnames';
 
-export default class Button extends Component {
-  render() {
-  	var { children, className, ...props } = this.props;
+const Button = ({className, children, ...props}) => (
```
Comment (gjjones) [inline src/components/Button/Button.js]: This was kept as the refactor tried to maintain existing code through the restructure. I'm not so cool with destructuring remaining properties into ```props```. ```props``` becomes the component properties minus anything already claimed. It feels a little misleading to say ```props``` but not have it represent all of the passed properties.

Comment (esimons) [inline src/components/Button/Button.js]: 100% agreed, and this isn't intended as a permanent pattern. I'm thinking we could namespace props that are meant to be generically destructured into element attributes (i.e. props.attr, or something like). Some kind of pattern that will differentiate between props that are intended to map to low-level HTML/SVGElement APIs vs props that are intended to map to higher-level component APIs.

**Comment by esimons** [comment]
You're a gentleman and a scholar.

File: src/components/Button/Button.js (hunk 1 above)
Comment (gjjones) [inline src/components/Button/Button.js]: I can dig it

**Pull request merged**